qemu/tests/check_reports_end_offset.py

```diff
@@ -0,0 +1,46 @@
+import logging
+import re
+import json
+
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+def run(test, params, env):
+    """
+    'qemu-img check' should report the end offset of the image.
+    1. Create a test qcow2 image.
+    2. Check there is image end offset and the value with both "humam"
+        and "json" output.
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def _check_result(key, offset, dict):
+        """Check the keywords and the value from the output."""
+        if key not in dict.keys() or int(offset) != int(dict[key]):
+            test.fail("The keyword/value is no correct. Check please.")
+
+    foo = params["images"]
+    root_dir = data_dir.get_data_dir()
+    foo = QemuImg(params.object_params(foo), root_dir, foo)
+    offset = params["image_end_offset"]
+    human_key = params["human_key"]
+    json_key = params["json_key"]
+
+    logging.info("Create the test image file.")
+    foo.create(foo.params)
+
+    """'qemu-img check' the image and check the output info."""
+    check_result = foo.check_image(foo.params, root_dir,
+                                   output="human").stdout.decode()
```

Comment:
the `check_result` might be `None` or an empty string in your current modification of `QemuImg.check_image`, you should take this into consideration.